exchanges/execute/src/execute.ts

```diff
@@ -0,0 +1,67 @@
+import { pipe, share, filter, map, fromPromise, mergeMap, merge } from 'wonka';
+import {
+  GraphQLSchema,
+  GraphQLFieldResolver,
+  GraphQLTypeResolver,
+  execute,
+} from 'graphql';
+import { Exchange, makeResult, makeErrorResult } from '@urql/core';
+
+interface ExecuteExchangeArgs {
+  schema: GraphQLSchema;
+  rootValue?: any;
+  contextValue?: any;
+  fieldResolver?: GraphQLFieldResolver<any, any>;
+  typeResolver?: GraphQLTypeResolver<any, any>;
+}
+
+const SUPPORTED_OPERATION_TYPES = ['query', 'mutation'];
+
+/** Exchange for executing queries locally on a schema using graphql-js. */
+export const executeExchange = ({
+  schema,
+  rootValue,
+  contextValue,
+  fieldResolver,
+  typeResolver,
+}: ExecuteExchangeArgs): Exchange => ({ forward }) => {
+  return ops$ => {
+    const sharedOps$ = share(ops$);
+
+    const executedOps$ = pipe(
+      sharedOps$,
+      filter(operation =>
+        SUPPORTED_OPERATION_TYPES.includes(operation.operationName)
+      ),
+      map(async operation => {
+        try {
+          const result = await execute(
```

Comment:
The last thing that's kind of missing is cancellation via the `teardown` operations; we can pair on this later, but generally we need to ensure that the result isn't issued when it's been cancelled, specifically for React Suspense 🙌